cpp/open3d/t/geometry/VtkUtils.cpp

```diff
@@ -0,0 +1,333 @@
+
+    const int cell_size = tensor.GetShape()[1];
+
+    auto tensor_flat = tensor.Reshape({tensor.NumElements(), 1}).Contiguous();
+    copy = copy && tensor.GetDataPtr() == tensor_flat.GetDataPtr();
+    auto connectivity = CreateVtkDataArrayFromTensor(tensor_flat, copy);
+
+    // vtk nightly build (9.1.20220520) has a function cells->SetData(cell_size,
+    // connectivity) which allows to remove the code below
+    vtkSmartPointer<vtkDataArray> offsets;
+    {
+        offsets.TakeReference(connectivity->NewInstance());
+        offsets->SetNumberOfTuples(1 + connectivity->GetNumberOfTuples() /
+                                               cell_size);
+
+        GenerateOffsetsImpl worker{cell_size,
+                                   connectivity->GetNumberOfTuples()};
+        using SupportedArrays = vtkCellArray::InputArrayList;
+        using Dispatch = vtkArrayDispatch::DispatchByArray<SupportedArrays>;
+        Dispatch::Execute(offsets, worker);
+    }
+    //--
+
+    vtkSmartPointer<vtkCellArray> cells = vtkSmartPointer<vtkCellArray>::New();
+    // grr, this always makes a deep copy.
+    // See ShallowCopy() in Common/Core/vtkDataArray.cxx for why.
+    cells->SetData(offsets, connectivity);
+
+    return cells;
+}
+
+// map types used by vtk to compatible Tensor types
+template <class T>
+struct VtkToTensorType {
+    typedef T TensorType;
+};
+
+template <>
+struct VtkToTensorType<long long> {
+    typedef int64_t TensorType;
+};
+
+struct CreateTensorFromVtkDataArrayWorker {
+    bool copy;
+    vtkSmartPointer<vtkDataArray> data_array;
+    core::Tensor result;
+
+    template <class TArray>
+    void operator()(TArray* array) {
+        typedef typename TArray::ValueType T;
+        typedef typename VtkToTensorType<T>::TensorType TTensor;
+        auto dtype = core::Dtype::FromType<TTensor>();
+        int64_t length = array->GetNumberOfTuples();
+        int64_t num_components = array->GetNumberOfComponents();
+
+        // copy if requested or the layout is not contiguous
+        if (copy || !array->HasStandardMemoryLayout()) {
+            result = core::Tensor({length, num_components}, dtype);
+            if (array->HasStandardMemoryLayout()) {
+                memcpy(array->GetVoidPointer(0), result.GetDataPtr(),
+                       dtype.ByteSize() * result.NumElements());
+            } else {
+                TTensor* data = result.GetDataPtr<TTensor>();
+                for (int64_t i = 0; i < length; ++i) {
+                    for (int64_t j = 0; j < num_components; ++j) {
+                        *data = array->GetTypedComponent(i, j);
+                        ++data;
+                    }
+                }
+            }
+        } else {
+            auto sp = data_array;
+            auto blob = std::make_shared<core::Blob>(core::Device(),
+                                                     array->GetVoidPointer(0),
+                                                     [sp](void*) { (void)sp; });
```

Comment:
Capturing `data_array` by value directly in the deleter lambda should work. We can skip `sp`.